spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -448,10 +449,93 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
     this
   }
 
+  @deprecated("Please use clone instead", "0.5.0")
   def cloneModule(): AbstractModule[A, B, T] = {
     SerializationUtils.clone(this)
   }
 
+  def clone(deepCopy : Boolean = true): AbstractModule[A, B, T] = {
+    val moduleData = ModuleData[T](this.
+      asInstanceOf[AbstractModule[Activity, Activity, T]], Seq[String](), Seq[String]())
+    val storages = new mutable.HashMap[Int, Any]()
+    val context = SerializeContext(moduleData, storages, ProtoStorageType, false)
+    val serializedModule = ModuleSerializer.serialize[T](context)
+      .bigDLModule
+    ModulePersister.setTensorStorage(serializedModule, storages)
+
+    storages.clear()
+
+    val deserializeContext = DeserializeContext(serializedModule.build,
+      storages, ProtoStorageType, false)
+    ModuleLoader.initTensorStorage[T](deserializeContext)
+    val copy = ModuleSerializer.load[T](deserializeContext).module
+      .asInstanceOf[AbstractModule[A, B, T]]
+    setWeightAndBias(copy, deepCopy)
+    copy
+  }
+
+
+  private def resetStorage(storages: mutable.HashMap[Int, Any]):
+    mutable.HashMap[Int, Any] = {
+    val storages = new mutable.HashMap[Int, Any]()
+    storages
```

Comment:
Two questions. 1. Why `reset` will return a new map here? 2. Have we used this method `resetStorage`?